com.unity.shadergraph/Editor/Drawing/Controllers/BlackboardCategoryController.cs

```diff
@@ -0,0 +1,298 @@
+            {
+                case AddShaderInputAction addBlackboardItemAction:
+                    if (IsInputInCategory(addBlackboardItemAction.shaderInputReference))
+                    {
+                        var blackboardRow = FindBlackboardRow(addBlackboardItemAction.shaderInputReference);
+                        if (blackboardRow == null)
+                            blackboardRow = InsertBlackboardRow(addBlackboardItemAction.shaderInputReference);
+                        // Rows should auto-expand when an input is first added
+                        // blackboardRow.expanded = true;
+                        var propertyView = blackboardRow.Q<SGBlackboardField>();
+                        if (addBlackboardItemAction.addInputActionType == AddShaderInputAction.AddActionSource.AddMenu)
+                            propertyView.OpenTextEditor();
+                    }
+                    break;
+
+                case CopyShaderInputAction copyShaderInputAction:
+                    if (IsInputInCategory(copyShaderInputAction.copiedShaderInput))
+                        InsertBlackboardRow(copyShaderInputAction.copiedShaderInput, copyShaderInputAction.insertIndex);
+                    break;
+
+                case AddItemToCategoryAction addItemToCategoryAction:
+                    // If item was added to category that this controller manages, then add blackboard row to represent that item
+                    if (addItemToCategoryAction.categoryGuid == ViewModel.associatedCategoryGuid)
+                    {
+                        InsertBlackboardRow(addItemToCategoryAction.itemToAdd, addItemToCategoryAction.indexToAddItemAt);
+                    }
+                    else
+                    {
+                        // If the added input has been added to a category other than this one, and it used to belong to this category,
+                        // Then cleanup the controller and view that used to represent that input
+                        foreach (var key in m_BlackboardItemControllers.Keys)
+                        {
+                            var blackboardItemController = m_BlackboardItemControllers[key];
+                            if (blackboardItemController.Model == addItemToCategoryAction.itemToAdd)
+                            {
+                                RemoveBlackboardRow(addItemToCategoryAction.itemToAdd);
+                                break;
+                            }
+                        }
+                    }
+                    break;
+
+                case DeleteCategoryAction deleteCategoryAction:
+                    if (deleteCategoryAction.categoriesToRemoveGuids.Contains(ViewModel.associatedCategoryGuid))
+                    {
+                        this.Destroy();
+                        return;
+                    }
+
+                    // Check if any inputs were added to this category (i.e. if a category merge took place)
```

Comment:
Can't merge categories anymore so see if this code is still relevant, remove if not

Reply:
removing code, can't have items added to a category on a category delete anymore